hack/install-wrapper.ps1

```diff
@@ -0,0 +1,194 @@
+{
+    $version = [System.Environment]::OSVersion.Version
+    $Script:OSVersion = $version.Major.ToString() + "." + $version.Minor.ToString() + "." + $version.Build.ToString()
+}
+
+function InitCloud()
+{
+    $version = (Get-WmiObject -class Win32_ComputerSystemProduct -namespace root\CIMV2).Version
+    $vendor = (Get-WmiObject -class Win32_ComputerSystemProduct -namespace root\CIMV2).Vendor
+    if (($version -like "*amazon") -or ($vendor -like "amazon*")) {
+        $script:Cloud = $AWS
+    } elseif ($vendor -like "*Microsoft*Corporation") {
+        $script:Cloud = $AZURE
+    } else {
+        Log "Unknown cloud platform. Only AWS and Azure Clouds are supported"
+        ExitOnError
+    }
+}
+
+function UpdateAntreaURL()
+{
+    $branchTag = $AntreaVersion.Substring(1,3)
+    $Script:AntreaBranch = "release-${branchTag}"
+    $Script:AntreaInstallScript = "https://raw.githubusercontent.com/antrea-io/antrea/${AntreaBranch}/hack/externalnode/install-vm.ps1"
+    $Script:AntreaConfig = "https://raw.githubusercontent.com/antrea-io/antrea/${AntreaBranch}/build/yamls/externalnode/conf/antrea-agent.conf"
+    $Script:AgentBin = "https://github.com/antrea-io/antrea/releases/download/${AntreaVersion}/antrea-agent-windows-x86_64.exe"
+}
+
+function GenerateNodename()
+{
+    $name = $null
+    $retries = 5
+    if ($Cloud -eq $AWS) {
+        for ($i=0; $i -lt $retries; $i++) {
+            try {
+                $endPoint = "$AWSMetaRoot/instance-id"
+                # NodeName is represented as vm-<instance-id>
+                $name = "vm-" + (Invoke-WebRequest -uri $endPoint).Content
+                break
+            } catch {
+                Log "Retrying query for $endPoint, attempt=$i"
+                Start-Sleep -Seconds 5
+                continue
+            }
+        }
+    } elseif ($Cloud -eq $AZURE) {
+        $vmId=""
+        $vmName=""
+        for ($i=0; $i -lt $retries; $i++) {
+            try {
+                $endPoint = "$AzureMetaRoot/instance/compute/resourceId?api-version=2021-01-01&format=text"
+                $vmId = (Invoke-RestMethod -Headers @{"Metadata"="true"} -Method GET -Uri $endPoint).ToLower()
+                $endPoint = "$AzureMetaRoot/instance/compute/name?api-version=2021-01-01&format=text"
+                $vmName = (Invoke-RestMethod -Headers @{"Metadata"="true"} -Method GET -Uri $endPoint).ToLower()
+                break
+            } catch {
+                Log "Retrying query for $endPoint, attempt=$i"
+                Start-Sleep -Seconds 5
+                continue
+            }
+        }
+        # Convert to ASCII
+        $vmIdAscii = [int[]][char[]]$vmId
+        # Compute sum of ASCII values of vmId
+        $sum=0
+        foreach ($i in $vmIdAscii) {
+            $sum = $sum + $i
+        }
+        # NodeName is represented as vm-<vm-name>-<hash of the VM resource ID>
+        $name = "vm-" + $vmName + "-" + $sum
+    }
+    $Script:Nodename = $name
+    if ($Nodename -eq "") {
```

Comment:
$name is possible configured with "$null" if any error happens, in this case the if condtion is not matched?

Reply:
Good catch. Will update it.